packages/server/src/adapters/node-http/nodeHTTPRequestHandler.ts

```diff
@@ -27,9 +27,17 @@ export async function nodeHTTPRequestHandler<
       return await opts.createContext?.(opts);
     };
 
-    const query = opts.req.query
-      ? new URLSearchParams(opts.req.query as any)
-      : new URLSearchParams(opts.req.url!.split('?')[1]);
+    // a framework could pre-parse query in a way that could crash URLSearchParams
+    const paramsFromQuery = () =>
+      new URLSearchParams(opts.req.url?.split('?')[1]);
+    let query: URLSearchParams;
+    try {
+      query = opts.req.query
+        ? new URLSearchParams(opts.req.query as any)
+        : paramsFromQuery();
+    } catch (err) {
+      query = paramsFromQuery();
+    }
```

Comment:
I think I do understand the need for this PR, but this feels a bit sub-optimal since there could be cases where *every* procedure ends up throwing (from `new URLSearchParams(opts.req.query as any)`) before being recovered by the `catch`.

It might be a little nitpicky but aren't there alternative implementations that wouldn't be require this? I don't want to speak for KATT here but maybe a boolean option on the server configuration instead of a `try/catch` on every procedure call?

Reply:
@fallenfreq friendly bump

Reply:
@Sheraff @juliusmarminge Hey, I've been camping with no signal for a few weeks and now I'm on a work trip for the next two weeks. I'll be able to look at this again when I'm home towards the end of the week starting July the 10th.

I didn't consider an extra parameters originally because in my case, it would only fail on a rare occurrences but it makes sense considering there are cases where it could fail on every request.

Would you have the parameter and keep the try catch block for cases where it has not been set? Or perhaps catch and re throw an error letting the user know they should be setting the parameter?

How do the names isQueryInvalid or invalidTrpcQuery sound?